Fix #155022 rst to markdown conversion

j-silv opened the pull request:
Used [rst2myst tool](https://rst-to-myst.readthedocs.io/en/latest/)

Fixes #155022

Docs comparison (check out the 'new' whenever docs build)

1. func.ux_limitations ([old](https://docs.pytorch.org/docs/main/func.ux_limitations.html) vs. [new](https://docs-preview.pytorch.org/pytorch/pytorch/155540/func.ux_limitations.html)) 
2. func.whirlwind_tour ([old](https://docs.pytorch.org/docs/main/func.whirlwind_tour.html) vs. [new](https://docs-preview.pytorch.org/pytorch/pytorch/155540/func.whirlwind_tour.html))
3. future_mod ([old](https://docs.pytorch.org/docs/main/future_mod.html) vs. [new](https://docs-preview.pytorch.org/pytorch/pytorch/155540/future_mod.html))
4. futures ([old](https://docs.pytorch.org/docs/main/futures.html) vs. [new](https://docs-preview.pytorch.org/pytorch/pytorch/155540/futures.html))
5. fx.experimental ([old](https://docs.pytorch.org/docs/main/fx.experimental.html) vs. [new](https://docs-preview.pytorch.org/pytorch/pytorch/155540/fx.experimental.html))

cc @svekars @sekyondaMeta @AlannaBurke

j-silv committed:
Fix #155022 rst to markdown conversion

Only git mv func.ux_limitations, func.whirlwind_tour, and futures
because after rst-markdown conversion, git thinks we deleted and added
new files. The conversion changes will come in a second commit/PR.

j-silv commented:
@pytorchbot label "topic: docs"

svekars commented:
These pages have Created On changed to Jun 09, 2025, this needs to be fixed:
https://docs-preview.pytorch.org/pytorch/pytorch/155540/func.ux_limitations.html
https://docs-preview.pytorch.org/pytorch/pytorch/155540/func.whirlwind_tour.html
https://docs-preview.pytorch.org/pytorch/pytorch/155540/futures.html

j-silv commented:
> These pages have Created On changed to Jun 09, 2025, this needs to be fixed: https://docs-preview.pytorch.org/pytorch/pytorch/155540/func.ux_limitations.html https://docs-preview.pytorch.org/pytorch/pytorch/155540/func.whirlwind_tour.html https://docs-preview.pytorch.org/pytorch/pytorch/155540/futures.html

How exactly can we fix this? I did use `git mv` but the changes between the .rst and .md files were too large and thus Git detected it as a delete/new file operation instead of a rename/modify. This is [happening here](https://github.com/pytorch/pytorch/pull/155120#issuecomment-2943883096).

I don't see how you can get Git to think it is the same file unless you modify the file with multiple commits... and since git thinks it's a new file, the "Created On" time which is extracted from git log will not track the original file.

j-silv committed:
Fix #155022 with previously renamed rst files

Split into 2 commits (this is the second one) so that we maintain
the git log of these files.
